### PR TITLE
Update incentive function

### PR DESCRIPTION
**Pull request by dleard**
- Fix ciip_incentive function to use real electricity / heat allocation values
- Add an id column to return type for unique key
- Coalesce incentive for product to equal 0 when no benchmark exists
- Add importedElectricityAllocationFactor & importedHeatAllocationFactor values to form_result dummy data.
- Change front-end to use incentive function instead of view
- Turn ciip_incentive function into a computed column on application_revision

### CODE REVIEW

File: schema/deploy/ciip_functions/ciip_incentive.sql [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 begin;
 -- Function takes application id and version number:
-create or replace function ggircs_portal.ciip_incentive(app_id int , version_no int )
+create or replace function ggircs_portal.ciip_incentive(app_id text , version_no text)
```
Comment (matthieu-foucault) [inline schema/deploy/ciip_functions/ciip_incentive.sql]: That will do for now. Ultimately, this should be a computed field of `applicationRevision`

File: schema/deploy/ciip_functions/ciip_incentive.sql [hunk 2]
```diff
@@ -42,29 +41,37 @@ returns setof ggircs_portal.ciip_incentive_by_product as $function$
...
      -- Loop over products
      for product in select * from ggircs_portal.ciip_production
-                    where version_number = version_no and application_id = app_id
+                    where version_number = version_no::integer and application_id = app_id::integer
      loop
 
+        -- If includes_imported_energy is false, heat/elec allocation factor=0 else set values from production view
+        em_alloc_elec = 0;
+        em_alloc_heat = 0;
+        if (select includes_imported_energy from ggircs_portal.benchmark b where b.product_id = product.product_id) then
```
Comment (matthieu-foucault) [inline schema/deploy/ciip_functions/ciip_incentive.sql]: You should use the `benchmark_data` below, which also filters by reporting year

File: schema/deploy/views/ciip_production.sql [hunk 3]
```diff
@@ -20,7 +20,9 @@ begin;
        (x.production_data ->> 'productUnits')::varchar(1000) as product_units,
        (x.production_data ->> 'productionAllocationFactor')::numeric as production_allocation_factor,
        (x.production_data ->> 'paymentAllocationFactor')::numeric as payment_allocation_factor,
-       (x.production_data ->> 'additionalData')::jsonb as additional_data
+       (x.production_data ->> 'additionalData')::jsonb as additional_data,
+       (x.production_data ->> 'importedElectricityAllocationFactor')::jsonb as imported_electricity_allocation_factor,
+       (x.production_data ->> 'importedHeatAllocationFactor')::jsonb as imported_heat_allocation_factor
```
Comment (matthieu-foucault) [inline schema/deploy/views/ciip_production.sql]: should these be cast as `numeric`?

Comment (dleard) [reply schema/deploy/views/ciip_production.sql]: yep!

File: schema/deploy/ciip_functions/ciip_incentive.sql (hunk 1 above)
Comment (dleard) [reply schema/deploy/ciip_functions/ciip_incentive.sql]: @matthieu-foucault  I'll go ahead and make it into a computed column

**Pull request merged**